packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -545,14 +565,13 @@ class ReactShallowRenderer {
           }
         }
 
-        if (element.type.hasOwnProperty('contextTypes')) {
+        if (elementType.hasOwnProperty('contextTypes')) {
```

Comment:
you may want to use Object.prototype.hasOwnProperty.call here, since it’d otherwise work fine to pass a function component with a null prototype.

Reply:
This codepath is for classes so should be ok. In fact it should probably use dot access since that's what we do in the real one.